pkg/build/builder/docker.go

```diff
@@ -53,9 +56,15 @@ func (d *DockerBuilder) Build() error {
 	if err != nil {
 		return err
 	}
-	if err := fetchSource(buildDir, d.build, d.urlTimeout, os.Stdin, d.git); err != nil {
+	sourceInfo, err := fetchSource(buildDir, d.build, d.urlTimeout, os.Stdin, d.git)
+	if err != nil {
 		return err
 	}
+	if sourceInfo != nil {
+		if err := updateBuildRevision(d.client, d.build, sourceInfo); err != nil {
```

Comment:
in general we probably don't want to fail the bulid just because we couldn't update this info anyway, though i'm less confident of that statement.